packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -319,15 +318,16 @@ class HierarchicalTermSelector extends Component {
 			const id = `editor-post-taxonomies-hierarchical-term-${ term.id }`;
 			return (
 				<div key={ term.id } className="editor-post-taxonomies__hierarchical-terms-choice">
-					<input
+					<CheckboxControl
 						id={ id }
-						className="editor-post-taxonomies__hierarchical-terms-input"
-						type="checkbox"
 						checked={ terms.indexOf( term.id ) !== -1 }
 						value={ term.id }
-						onChange={ this.onChange }
+						onChange={ () => {
```

Comment:
The previous approach was better, because it didn't create a new function on every render, and hence break equality checks.

Reply:
yes. but. the CheckboxControl hijacks the event of the click and returns only true or false and I need to get that id.

Reply:
Maybe we should change `CheckboxControl` to add the full event as the second parameter to `onChange`.

Reply:
Mind the mobile :) (events).

I think inline function are fine. We should optimize if we measure performance issues in that case. we can use things like `useCallback` if needed

Reply:
>Mind the mobile :) (events).

Good point.

>`useCallback`

Yes.